k2/csrc/cuda/debug_test.cu

```diff
@@ -0,0 +1,190 @@
+// k2/csrc/cuda/debug_test.cu
+
+// Copyright (c) 2020, Xiaomi Corporation ( authors: Meixu Song )
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#ifndef K2_PARANOID
+#define K2_PARANOID
+#endif
+
+#include "k2/csrc/cuda/debug.h"
+
+namespace k2 {
+
+// A dummy cuda asynchronous function
+__global__ void FillContents(int n, int *output) {
+  int correct_index = static_cast<int>(threadIdx.x) * n;
+
+  for (int i = correct_index; i < n; i++) {
+    output[i] = (i - correct_index) * 2;
+    K2_DLOG("Print from async cuda kernel, output[%d] = %d\n", i, output[i]);
+  }
+}
+
+/* number of blocks (max 65,536) */
+#define BLOCKS 50000
+
+/* number of threads per block (max 1,024) */
+#define THREADS 200
+
+/* N is now the number of blocks times the number of threads
+   in this case it is 10 million */
+#define N (BLOCKS * THREADS)
+
+/* the kernel fills an array up with square roots */
+__global__ void square_roots(double* array) {
```

Comment:
Please follow the code style and rename your function name.

Reply:
Please pay attention to where to put the star. It is `double *array`, not `double* array`.

Reply:
Done.